File: app/src/main/java/com/peakflow/you_must_not_cheat_yourself_your_parents_raised_a_soldier_prove_it/SplashScreen.java
package com.peakflow.you_must_not_cheat_yourself_your_parents_raised_a_soldier_prove_it;

import static kotlinx.coroutines.DelayKt.delay;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class SplashScreen extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_splash_screen);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        //
        new Handler(Looper.getMainLooper()).postDelayed(()->{
            Intent i = new Intent(SplashScreen.this, Login_Activity.class);
            startActivity(i);
            finish();
        }, 2000);

        SharedPreferences prefs = getSharedPreferences("MyAppPrefs", MODE_PRIVATE);
        String uid = prefs.getString("uid", null);

        if (uid != null) {
            // User is logged in, fetch details from Firebase
            fetchUserDetails(uid);
        } else {
            // Redirect to login screen
            startActivity(new Intent(SplashScreen.this, DashboardActivity.class).putExtra("uid", uid));

            finish();
        }

    }
    private void fetchUserDetails(String firebaseUid) {
        DatabaseReference userRef = FirebaseDatabase.getInstance().getReference("Users").child(firebaseUid);

        userRef.get().addOnCompleteListener(task -> {
            if (task.isSuccessful() && task.getResult().exists()) {
                User user = task.getResult().getValue(User.class);
                if (user != null) {
                    // ✅ Successfully fetched user details
                    Toast.makeText(this, "Welcome, " + user.name, Toast.LENGTH_SHORT).show();
                }
            } else {
                Toast.makeText(this, "User not found!", Toast.LENGTH_SHORT).show();
            }
        });
    }

}

File: app/src/main/java/com/peakflow/you_must_not_cheat_yourself_your_parents_raised_a_soldier_prove_it/SplashScreen.kt
package com.peakflow.you_must_not_cheat_yourself_your_parents_raised_a_soldier_prove_it

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import dagger.hilt.android.AndroidEntryPoint
import javax.inject.Inject
import androidx.lifecycle.ViewModel

@AndroidEntryPoint
class SplashScreen : AppCompatActivity() {
    private val viewModel: SplashViewModel by viewModels()
    private val handler = Handler(Looper.getMainLooper())
    
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)
        
        // Initialize notification channel
        NotificationHelper.createNotificationChannel(this)
        
        // Navigate to appropriate screen after delay
        handler.postDelayed({
            val intent = Intent(this, LoginActivity::class.java)
            startActivity(intent)
            finish()
        }, 2000) // 2 second delay
    }
    
    override fun onDestroy() {
        super.onDestroy()
        handler.removeCallbacksAndMessages(null)
    }
}

class SplashViewModel @Inject constructor() : ViewModel() {
    // Add any initialization logic here
} 